conseil-api/src/main/scala/tech/cryptonomic/conseil/api/routes/platform/data/ApiDataOperations.scala

```diff
@@ -78,24 +78,68 @@ trait ApiDataOperations extends DatabaseMetadataOperations with DataOperations {
       outputType: OutputType,
       limit: Int
   )(implicit ec: ExecutionContext): DBIO[List[QueryResponse]] = {
+    /* we add [pre/post]-processing of the queries to take into
+     * account the need to hide any fork-invalidated data
+     */
     val tableWithPrefix = prefix + "." + table
+    val effectiveColumns = columns.filterNot(hiddenForkFields)
+    val effectiveAggregations = aggregation.filterNot(hiddenForkFields)
+    val effectiveOrdering = ordering.filterNot(hiddenForkFields)
+    val effectivePredicates = hideForkResults(predicates) ++ predicates
     val q = (temporalPartition, snapshot) match {
       case (Some(tempPartition), Some(snap)) =>
-        makeTemporalQuery(tableWithPrefix, columns, predicates, aggregation, ordering, tempPartition, snap, limit)
+        makeTemporalQuery(
+          tableWithPrefix,
+          effectiveColumns,
+          effectivePredicates,
+          effectiveAggregations,
+          effectiveOrdering,
+          tempPartition,
+          snap,
+          limit
+        )
       case _ =>
-        makeQuery(tableWithPrefix, columns, aggregation)
-          .addPredicates(predicates)
-          .addGroupBy(aggregation, columns)
-          .addHaving(aggregation)
-          .addOrdering(ordering)
+        makeQuery(tableWithPrefix, effectiveColumns, effectiveAggregations)
+          .addPredicates(effectivePredicates)
+          .addGroupBy(effectiveAggregations, effectiveColumns)
+          .addHaving(effectiveAggregations)
+          .addOrdering(effectiveOrdering)
           .addLimit(limit)
     }
 
     if (outputType == OutputType.sql) {
       DBIO.successful(List(Map("sql" -> Some(q.queryParts.mkString("")))))
     } else {
-      q.as[QueryResponse].map(_.toList)
+      q.as[QueryResponse].map(_.toList.map(hideForkFields))
     }
   }
 
+  /** We collect here field names that only relates to the chain fork handling process.
+    * As such, these fields are not supposed to appear as results or to have an effect on
+    * any predicated query being run.
+    *
+    * Defaults to an empty set.
+    */
+  protected def forkRelatedFields: Set[String] = Set.empty
+
+  /** Generates extra predicates to skip whole records that would make the fork logic apparent in results.
+    * A common implementation will add a filter on the "invalidation flag/date" field to be empty or false.
+    *
+    * Defaults to an empty list.
+    */
+  protected def hideForkResults(userQueryPredicates: List[Predicate]): List[Predicate] = List.empty
```

Comment:
I'm not sure why it defaults to empty list. Are you going to add a predicate that checks if row was invalidated later?

Reply:
Maybe I misread the question.
The actual implementation will add an additional predicate for the "invalidated_asof" column to be null.

Such implementation is currently custom to the tezos schema. Other chains will use the default empty predicate list, so that the queries will not be impacted for now.
I assume that we'll later apply a similar forking algorithm to ethereum and bitcoin too, but I'm not doing that right now, for lack of details.

Reply:
oh, ok! My mind defaulted to the tezos only implementation. That explains a lot, thanks! 😄